.github/workflows/publish-oss.yml

```diff
@@ -44,41 +47,41 @@ defaults:
         shell: bash
 
 jobs:
+    bump-version:
+        if: inputs.version != ''
+        uses: ./.github/workflows/_bump-version.yml
+        with:
+            package: ${{ inputs.package }}
+            branch: ${{ inputs.branch }}
+            version: ${{ inputs.version }}
+        secrets: inherit
+
     unit-tests:
         if: |
+            always() && !failure() &&
             inputs.skip-unit-tests == false &&
             !contains(fromJSON('["dbt-tests-adapter"]'), inputs.package)
+        needs: bump-version  # bump-version can be skipped and is not directly used
         uses: ./.github/workflows/_unit-tests.yml
         with:
             package: ${{ inputs.package }}
             branch: ${{ inputs.branch }}
 
     integration-tests:
         if: |
+            always() && !failure() &&
             inputs.skip-integration-tests == false &&
             !contains(fromJSON('["dbt-adapters", "dbt-tests-adapter"]'), inputs.package)
+        needs: bump-version  # bump-version can be skipped and is not directly used
         uses: ./.github/workflows/_integration-tests.yml
         with:
             packages: ${{ toJSON(inputs.package) }}
             branch: ${{ inputs.branch }}
         secrets: inherit
 
-    publish-prep-checks:
-        name: "Publish prep checks"
-        if: always()
-        needs: [unit-tests, integration-tests]
-        runs-on: ${{ vars.DEFAULT_RUNNER }}
-        steps:
-        -   uses: re-actors/alls-green@release/v1
-            with:
-                jobs: ${{ toJSON(needs) }}
-                allowed-skips: ${{ toJSON(needs) }}
-
     generate-changelog:
-        needs: publish-prep-checks
-        if: |
-            always() &&
-            needs.publish-prep-checks.result == 'success'
+        if: always() && !failure()
```

Comment:
This is right - but makes me dislike the logic handling of workflows a little more each time I see it.

Reply:
All CI/CD tools are wrong, but some are useful.